src/utilities/geometry/Polygon3d.hpp

```diff
@@ -0,0 +1,117 @@
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef UTILITIES_GEOMETRY_POLYGON_HPP
+#define UTILITIES_GEOMETRY_POLYGON_HPP
+
+#include "../UtilitiesAPI.hpp"
+
+#include "Point3d.hpp"
+#include <vector>
+
+namespace openstudio {
+
+class Vector3d;
+
+class UTILITIES_API Polygon3d
+{
+ public:
+  // Constructs an empty polygon
+  Polygon3d();
+
+  // Construct a polygon with an outer path
+  Polygon3d(const Point3dVector& outerPath);
+
+  // Constructs a polygon with an outer path and one or more inner paths
+  Polygon3d(const Point3dVector& outerPath, const Point3dVectorVector& innerPaths);
+
+  // Assigns an outer path for the polygon
+  void setOuterPath(const Point3dVector& outerPath);
+
+  // Returns the polygon's outer path
+  Point3dVector getOuterPath() const;
+
+  // Returns the polygon's inner paths
+  Point3dVectorVector getInnerPaths() const;
+
+  // Adds a point to the outer path of the polygon
+  void addPoint(const Point3d& point);
+
+  // Adds an inner path to the polygon
+  void addHole(const Point3dVector& hole);
+
+  // Calculates the Newall Vector for the polygon
+  Vector3d newallVector();
+
+  // Calculates the outward normal of the polygon
+  Vector3d outwardNormal();
+
+  // Calculates the gross area of the polygon (area excluding holes)
+  double grossArea();
+
+  // Calculates the net area of the polygon (gross area - area of holes)
+  double netArea();
+
+  // Gets the perimeter of the outer path of the polygon
+  double perimeter();
+
+  // Determines whether the polygon os clockwise (normal down) or anti-clockwise (normal up)
+  bool isClockwise();
+
+  // Calculates the centroid of the polygon
+  Point3d centroid();
+
+  // Point is on the outer path or one of the inner paths
+  bool pointInPolygon(const Point3d& point, double tol = 0.01);
+
+  // Point inside outer path and not inside a hole
+  bool within(const Point3d& point, double tol = 0.01);
+
+  // Point is inside the outer path and not inside a hole or on the outer path or one of the inner paths
+  bool inside(const Point3d& point, double tol = 0.01);
+
+  // gets the size of the outer path
+  size_t getSize();
+
+  // gets an edge on thge outer path
+  Point3dVector getEdge(size_t i);
+
+  // Returns the ovelapping segments between this polygon and the line
+  std::vector<Point3dVector> overlap(const Point3dVector& line) const;
+
+ private:
+  // The polygon's outer path
+  Point3dVector m_outerPath;
```

Comment:
@jmarrec I would not hold this PR up for this comment, but do you think this class should follow PIMPL to hide these members? 

@macumber might have an opinion too. 

Regardless I think that could be done retroactively if we choose.

Reply:
I don't think we need Pimpl here. This is short-lived, for computing intersections and co, and isn't something like a ModelObject that we need to serialize and also query all objects like model.objects()

Reply:
I agree with @jmarrec here.  The other geometry classes are Plain Old Data types